src/vdom/component.js

```diff
@@ -17,39 +17,67 @@ import { removeNode } from '../dom/index';
  *	@param {boolean} [opts.render=true]			If `false`, no render will be triggered.
  */
 export function setComponentProps(component, props, opts, context, mountAll) {
-	if (component._disable) return;
-	component._disable = true;
 
-	if ((component.__ref = props.ref)) delete props.ref;
-	if ((component.__key = props.key)) delete props.key;
+	const requestIdleCallback =
+      typeof window !== 'undefined' &&
+      typeof window.requestIdleCallback !== 'undefined' ?
+         window.requestIdleCallback :
+         null;
 
-	if (!component.base || mountAll) {
-		if (component.componentWillMount) component.componentWillMount();
-	}
-	else if (component.componentWillReceiveProps) {
-		component.componentWillReceiveProps(props, context);
-	}
+	const cancelIdleCallback =
+      requestIdleCallback !== 'null' ?
```

Comment:
Nah, I didn't. I did a typeof test earlier on, and that carried through. Awwwwkward.